kinesis.go

```diff
@@ -85,29 +125,32 @@ func (k *kinesis) init(stream, shard, shardIteratorType, accessKey, secretKey, r
 	return k, nil
 }
 
-func (k *kinesis) args() *gokinesis.RequestArgs {
-	args := gokinesis.NewArgs()
-	args.Add("StreamName", k.stream)
-	args.Add("ShardId", k.shard)
-	args.Add("ShardIterator", k.shardIterator)
-
+func (k *kinesis) initShardIterator() error {
+	// log.Println(k.sequenceNumber)
+	var awsSeqNumber *string
 	if k.sequenceNumber != "" {
-		args.Add("StartingSequenceNumber", k.sequenceNumber)
-		args.Add("ShardIteratorType", ShardIterTypes[atSequenceNumber])
-	} else {
-		args.Add("ShardIteratorType", k.shardIteratorType)
+		awsSeqNumber = aws.String(k.sequenceNumber)
 	}
-
-	return args
-}
-
-func (k *kinesis) initShardIterator() error {
-	resp, err := k.client.GetShardIterator(k.args())
+	k.sequenceNumber = aws.StringValue(awsSeqNumber)
+	resp, err := k.client.GetShardIterator(&awsKinesis.GetShardIteratorInput{
+		ShardId:                aws.String(k.shard),             // Required
+		ShardIteratorType:      aws.String(k.shardIteratorType), // Required
+		StreamName:             aws.String(k.stream),            // Required
+		StartingSequenceNumber: awsSeqNumber,
+		Timestamp:              aws.Time(time.Now()),
+	})
+	log.Print("here1")
 	if err != nil {
+		log.Print("here2")
+
 		return err
 	}
+	if resp.ShardIterator != nil {
```

Comment:
Wouldn't it be easier to just change setShardIterator() to take a pointer to a string?

It is used another time in listener.go consume() which is the same in that we have the ShardIterator from the response as a pointer to String.